eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/permission/SinglePermission.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.permission;
+
+import android.Manifest;
+
+
+/**
+ * Represents a single Android permission. Like {@link Manifest.permission#ACCESS_FINE_LOCATION}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public interface SinglePermission
+{
+
+    /**
+     * Tells whether the permission is granted currently.
+     */
+    boolean granted();
+
+    /**
+     * Requests the permission from the user.
+     *
+     * @param listener
+     *         listener for notifying about the user's decision
+     */
+    void requestGrant(PermissionGrantListener listener);
```

Comment:
The Javadoc says it requests a "Permission", the name says it requests a "Grant"? I think it's sane to call this method just `request`.